src/HTTPConnection.cpp

```diff
@@ -133,6 +133,7 @@ void HTTPConnection::closeConnection() {
 
   if (_httpHeaders != NULL) {
     HTTPS_LOGD("Free headers");
+    _wsHandler->onClose(); 
     delete _httpHeaders;
     _httpHeaders = NULL;
   }
```

Comment:
Did you really mean to add `_wsHandler->onClose()` to this block? It can be `NULL` here. The block below checks if a websocket handler does exist and in that case deconstructs it:

```c++
  if (_wsHandler != nullptr) {
    HTTPS_LOGD("Free WS Handler");
    // Shouldn't the call to _wsHandler->onClose() go here?
    delete _wsHandler;
    _wsHandler = NULL;
  }
```

Reply:
No, I didn't. Actually in my version the _wsHandler->onClose  is exaclty at the place You suggested. It was a copy-paste error as I'm still learning the whole version control thing. This was the first time I created a PR so I did it 'by hand' (moreover this PR was the reason I registered on github).